extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -51,17 +53,20 @@ export class AmpAd extends AMP.BaseElement {
 
     return consent.then(() => {
       const type = this.element.getAttribute('type');
-      if (!type) {
-        // Unspecified or empty type.  Nothing to do here except bail out.
-        return null;
-      }
+      const isCustom = type === 'custom' && isExperimentOn(this.win,
+          'ad-type-custom');
+      user().assert(isCustom || hasOwn(adConfig, type)
```

Comment:
Validator requires `type` to be present, eliminating `null`. It can be any string, though.